src/component/todo-list/todo-list.ts

```diff
@@ -0,0 +1,109 @@
+import '../todo-item/todo-item';
+import { todoStore } from '../../utils/todo-store';
+import { filterStore } from '../../utils/filter-store';
+import { Todo } from '../../models/todo';
+import { FilterType } from '../../models/filter';
+
+export class TodoList extends HTMLElement {
+  private todoStore = todoStore;
+  private filterStore = filterStore;
+  private filter: FilterType = 'all';
+  private todoListData: Todo[] = [];
+  private sortFlag: boolean = true;
+
+  private updateTodoList = () => {
+    const todos = this.todoStore.getState();
+    this.filter = this.filterStore.getState();
+    this.todoListData = todos.filter(todo => {
+      if (this.filter === 'all') return true;
+      if (this.filter === 'active') return !todo.done;
+      if (this.filter === 'completed') return todo.done;
+      return true;
+    });
+
+    this.render();
+
+    this.dispatchEvent(
+      new CustomEvent('list-render', {
+        bubbles: true
+      })
+    );
+  };
```

Comment:
all일때는 반복문을 순회하지 않아도되기때문에 아래와 같이 리팩토링 해주었습니다. 

```suggestion
  private filterTodoList = () => {
    const todos = this.todoStore.getState();
    const filter = this.filterStore.getState();

    if (filter === 'active') return todos.filter(({ done }) => !done);

    if (filter === 'completed') return todos.filter(({ done }) => done);

    return todos;
  };

  private updateTodoList = () => {
    this.todoListData = this.filterTodoList();

    this.render();

    this.dispatchEvent(
      new CustomEvent('list-render', {
        bubbles: true
      })
    );
  };
```